src/librustc/ty/context.rs

```diff
@@ -1672,6 +1672,7 @@ macro_rules! nop_list_lift {
 }
 
 nop_lift!{Ty<'a> => Ty<'tcx>}
+nop_lift!{&'a LayoutDetails => &'tcx LayoutDetails}
```

Comment:
Do you still need this? Also, why are all of these called `nop` when these are the base impls that do *all* of the lifting work?!

Reply:
They are called `nop` because they don't do any cloning + interning, they bail out if the thing does not exist in the interner yet.

I think this naming may come from a complete misunderstanding of lifting on my side, because nowadays (if ever) we don't do the clone + interning thing.

Reply:
Lifting should *never* clone + intern! (although it might need to allocate new `Vec`s etc.)

It can fail precisely because it looks for already allocated/interned things, and they might not be there.

They should probably be called `lift_to_arena!` or something.

Reply:
These have all been removed